octobot_trading/personal_data/portfolios/history/historical_portfolio_value_manager.py

```diff
@@ -163,13 +166,16 @@ def _add_historical_portfolio_value(self, timestamp, value_by_currency):
             historical_asset_value.HistoricalAssetValue(timestamp, value_by_currency)
 
     def _update_portfolios(self):
+        if self.portfolio_manager.portfolio is None or self.portfolio_manager.portfolio.portfolio is None:
+            return
         self.ending_portfolio = portfolio_util.portfolio_to_float(
             self.portfolio_manager.portfolio.portfolio
         )
         if self.starting_portfolio is None:
-            if self.portfolio_manager.portfolio_value_holder.origin_portfolio is None:
+            if self.portfolio_manager.portfolio_value_holder.origin_portfolio is None \
+                    or not self.portfolio_manager.portfolio_value_holder.origin_portfolio.portfolio:
                 # origin portfolio might not be initialized, use ending_portfolio
-                self.starting_portfolio = self.ending_portfolio
+                self.starting_portfolio = copy.deepcopy(self.ending_portfolio)
```

Comment:
Why didn't we do that before?

Reply:
I forgot but it's necessary otherwise we will change the starting portfolio if we update the current one